plugins/inputs/azure_monitor/azure_monitor.go

```diff
@@ -59,8 +63,20 @@ func (am *AzureMonitor) SampleConfig() string {
 
 // Init is for setup, and validating config.
 func (am *AzureMonitor) Init() error {
+	var clientOptions azcore.ClientOptions
+	switch cloudOption := am.CloudOption; cloudOption {
+	case "AzureChina":
+		clientOptions = azcore.ClientOptions{Cloud: cloud.AzureChina}
+	case "AzureGovernment":
+		clientOptions = azcore.ClientOptions{Cloud: cloud.AzureGovernment}
+	case "AzurePublic":
+		clientOptions = azcore.ClientOptions{Cloud: cloud.AzurePublic}
+	default:
+		clientOptions = azcore.ClientOptions{Cloud: cloud.AzurePublic}
```

Comment:
```suggestion
	case "", "AzurePublic":
		clientOptions = azcore.ClientOptions{Cloud: cloud.AzurePublic}
	default:
		return fmt.Errorf("Unknown cloud option: %s", cloudOption)
```

Reply:
@powersj Thank you for the information. I've adapted the part.